### PR TITLE
Default leaveOpen to false in ODataBinaryStreamValue constructor

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
### Issues

*This pull request fixes https://github.com/OData/odata.net/issues/2599*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Core/Value/ODataBinaryStreamValue.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public sealed class ODataBinaryStreamValue : ODataValue
         /// </summary>
         /// <param name="stream">Input stream</param>
         public ODataBinaryStreamValue(Stream stream) :
-            this(stream, true) // Leaves the stream open by default, for backward compatibility.
+            this(stream, false) // Leaves the stream open by default, for backward compatibility.
```
Comment (habbes) [inline src/Microsoft.OData.Core/Value/ODataBinaryStreamValue.cs:23]:
```suggestion
            this(stream, leaveOpen: false)

```

File: src/Microsoft.OData.Core/Value/ODataBinaryStreamValue.cs [hunk 2]
```diff
@@ -20,7 +20,7 @@ public sealed class ODataBinaryStreamValue : ODataValue
         /// </summary>
         /// <param name="stream">Input stream</param>
         public ODataBinaryStreamValue(Stream stream) :
-            this(stream, true) // Leaves the stream open by default, for backward compatibility.
+            this(stream, leaveOpen: false)
```
Comment (mikepizzo) [inline src/Microsoft.OData.Core/Value/ODataBinaryStreamValue.cs:23]: >leaveOpen: false [](http://example.com/codeflow?start=25&length=16)

are there no tests that need to be updated by changing this default? can we add tests to verify this default?

File: test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedSerializerUndecalredTests.cs [hunk 3]
```diff
@@ -235,6 +235,29 @@ public void WriteResourceCollectionUntypedProperty_WorksResourceInNestedCollecti
             Assert.Equal(result, actual);
         }
 
+        [Fact]
+        public void ODataBinaryStreamValue_Leaves_Stream_Not_Open_By_Default()
```
Comment (habbes) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedSerializerUndecalredTests.cs:239]:
```suggestion
        public void ODataBinaryStreamValue_Closes_Stream_By_Default()

```

File: test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedSerializerUndecalredTests.cs [hunk 4]
```diff
@@ -235,6 +235,29 @@ public void WriteResourceCollectionUntypedProperty_WorksResourceInNestedCollecti
...
+                Name = "Data",
+                Value = new ODataBinaryStreamValue(stream)
+            };
+
+            string result = WriteDeclaredUntypedProperty(property);
+            Assert.Equal("{\"@odata.context\":\"http://www.sampletest.com/$metadata#serverEntitySet/$entity\",\"Data\":\"CjEyMzQ1Njc4OTA=\"}", result);
+
+            Assert.False(stream.CanRead);
+        }
+
```
Comment (habbes) [inline test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedSerializerUndecalredTests.cs:260]: Is there a test that shows if you set `leaveOpen: true` that it won't automatically close the stream? If not, I think you should add that as well.

Comment (ElizabethOkerio) [reply test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightEntryAndFeedSerializerUndecalredTests.cs:260]: It's there: https://github.com/OData/odata.net/blob/main/test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightValueSerializerTests.cs#L320

**Review by habbes — APPROVED** [review]

**Review by gathogojr — APPROVED** [review]

**Pull request merged**